### PR TITLE
Vote example

### PR DESCRIPTION
**Pull request by eysi09**
Need to squash before we merge.

### CODE REVIEW

File: examples/vote/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Voting example project
+
+Example voting application where you can vote for either cats or dogs. You can vote as many times as you would like.
```
Comment (edvald) [inline examples/vote/README.md]: Maybe add something about the intent here, and the structure of the application. What we're demonstrating is a fairly typical backend application configured with Garden, as opposed to how to make a voting application.

File: examples/vote/services/frontend/public/index.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1.0">
+    <title>hello-world</title>
+  </head>
+  <body>
+    <noscript>
+      <strong>We're sorry but hello-world doesn't work properly without JavaScript enabled. Please enable it to continue.</strong>
```
Comment (edvald) [inline examples/vote/services/frontend/public/index.html]: hello-world?

File: examples/vote/services/postgres/garden.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        - name: data
+          containerPath: /db-data
+      ports:
+        - name: db
+          containerPort: 5432
+      healthCheck:
+        command: [psql, -w, -U, postgres, -d, postgres, -c, "SELECT 1"]
+  tasks:
+    - name: db-init
+      command: [psql, -w, -U, postgres, --host=db, --port 5432, -d, postgres, "-c 'CREATE TABLE IF NOT EXISTS votes (id VARCHAR(255) NOT NULL UNIQUE, vote VARCHAR(255) NOT NULL, created_at timestamp default NULL)'"]
```
Comment (edvald) [inline examples/vote/services/postgres/garden.yml]: This is a little ugly maybe... I guess adding a script to the postgres container wouldn't make much sense, but what about making a basic migration module, or adding a script to one of the service containers?

File: examples/vote/services/vote/garden.yml [hunk 4]
```diff
@@ -6,11 +6,15 @@ module:
...
           port: interface
       ports:
         - name: interface
           protocol: TCP
           containerPort: 80
       dependencies:
         - redis
+  tests:
+    - name: unit
+      command: [echo ok]
```
Comment (edvald) [inline examples/vote/services/vote/garden.yml]: A comment here would be nice (just explain that we're demonstrating how to define tests, and less how to write unit tests). Also, should be `[echo, ok]`, note the comma.

File: examples/vote/services/result/server.js [hunk 5]
```diff
@@ -73,8 +75,12 @@ app.use(function(req, res, next) {
   next();
 });
 
-app.use(express.static(__dirname + '/views'));
+app.use(express.static('views'));
+// app.use(express.static(__dirname + '/result/views'));
 
+// app.use('/result', express.static(__dirname + '/views'));
+
+// app.get('/result', function (req, res) {
```
Comment (edvald) [inline examples/vote/services/result/server.js]: Clean up commented lines (unless these are used to debug or something?)

File: examples/vote/services/result/garden.yml [hunk 6]
```diff
@@ -5,10 +5,18 @@ module:
...
         - path: /
           port: ui
+          hostname: results.local.app.garden
       ports:
         - name: ui
           protocol: TCP
           containerPort: 80
+  tests:
+    - name: integ
+      command: [echo ok]
```
Comment (edvald) [inline examples/vote/services/result/garden.yml]: Would be really good to have actual integ tests

File: examples/vote/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+✔ result                    → Checking status → Version 8bbc389b3e already deployed
+```
+
+### To Vote
+
+open http://vote.local.app.garden/
+
+### View Results
+
+open http://vote.local.app.garden/result
```
Comment (edvald) [inline examples/vote/README.md]: Is this accurate?

File: examples/vote/services/frontend/garden.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    - name: frontend
+      command: [npm, run, serve]
+      hotReloadCommand: [npm, run, serve]
+      ports:
+        - name: http
+          containerPort: 8080
+      ingresses:
+        - path: /
+          port: http
+          hostname: frontend.local.app.garden
```
Comment (edvald) [inline examples/vote/services/frontend/garden.yml]: Is the naming maybe a little confusing? `vote`, `results` and `frontend`... From the outside I'd expect `vote` to be the voting UI, `results` to be the results UI and then "frontend" would be...? Either needs explanation or a little more clarity in the naming.

File: examples/vote/garden.sh [hunk 9]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+GARDEN_ENABLE_SERVER=1 garden dev --hot-reload=vote
```
Comment (drubin) [inline examples/vote/garden.sh]: Maybe also include `GARDEN_SERVER_PORT=9777` also please do `#!/bin/bash -e` :)

**Pull request merged**